sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -435,7 +435,7 @@ _pi_context::getFreeSlotInExistingOrNewPool(ze_event_pool_handle_t &Pool,
                                             size_t &Index, bool HostVisible,
                                             bool ProfilingEnabled) {
   // Lock while updating event pool machinery.
-  std::scoped_lock Lock(ZeEventPoolCacheMutex);
+  std::scoped_lock<pi_mutex> Lock(ZeEventPoolCacheMutex);
```

Comment:
Why was this needed ? The `pi_mutex` is dervide from std:mutex. Tag @againull who authored this

Reply:
It seems this was done in https://github.com/intel/llvm/pull/6748/commits/afbc09286a832ef04cc55020b4c548fe2dfb051d to silence the -Wctad-maybe-unsupported warning, i.e. not a functional requirement.

Reply:
Correct. std::mutex may work here as well if that's better.

Reply:
I don't think it's better.

Reply:
The "right" way (according to community) to fix that would be to add explicit deduction guides to headers, but that would be mutex header, which is a system (libstdc++) header. This is sort of workaround until system headers do that
... or community agrees that system headers must be excluded (https://reviews.llvm.org/D133425).
Tagging @AaronBallman

Reply:
I am OK with the proposed change, just need to make sure that future cases like this follow it as well. When is the warning triggered? in which build?

Reply:
Warning is added into LLVM build system inside this pulldown.
As soon as we complete this pulldown, pre-commit CI should fail on build if this is not done (under -Werror).
When building locally you need to add --werror to configure.py.

Reply:
> pre-commit CI should fail on build if this is not done

Sounds good.